pkg/distributor/distributor_test.go

```diff
@@ -3751,15 +3746,44 @@ func prepareRingInstances(cfg prepConfig, ingesters []*mockIngester) *ring.Desc
 	return &ring.Desc{Ingesters: ingesterDescs}
 }
 
-func prepare(t testing.TB, cfg prepConfig) ([]*Distributor, []*mockIngester, []*prometheus.Registry) {
+func preparePartitionsRing(cfg prepConfig, ingesters []*mockIngester) *ring.PartitionRingDesc {
+	desc := ring.NewPartitionRingDesc()
+
+	// When we add partitions we simulate the case they were switched to ACTIVE older
+	// than the shuffle sharding lookback period, so that we don't query all partitions
```

Comment:
```suggestion
	// When we add partitions we simulate the case they were switched to ACTIVE before
	// the shuffle sharding lookback period, so that we don't extend partitions
```

another nit

Reply:
Done in https://github.com/grafana/mimir/pull/7376/commits/bc54fcdafe8dd243f94d10da8c18b6c511589394